contracts/Removal.sol

```diff
@@ -29,22 +37,44 @@ contract Removal is
 
   struct BatchMintRemovalsData {
     // todo why doesnt typechain generate this as a type?
+    uint256 projectId;
+    uint256 scheduleStartTime;
     address marketAddress;
     bool list;
   }
 
-  uint256 private _tokenIdCounter;
+  struct ScheduleData {
+    uint256 startTime;
+    address supplierAddress;
+    uint256 methodology;
+    uint256 methodologyVersion;
+  }
+
+  /**
+   * @notice The RestrictedNORI contract that manages restricted tokens.
+   */
+  RestrictedNORI private _restrictedNori;
+  uint256 public tokenIdCounter;
   string public name; // todo why did I add this
   mapping(uint256 => uint256) public indexToTokenId; // todo consider how we're keeping track of the number and order of ids, ability to iterate
   mapping(uint256 => bool) private _tokenIdExists;
+  mapping(uint256 => uint256) private _removalIdToProjectId;
+  mapping(uint256 => ScheduleData) private _projectIdToScheduleData;
 
-  function initialize() public virtual initializer {
+  function initialize() external virtual initializer {
     super.initialize("https://nori.com/api/removal/{id}.json");
     __ERC1155Supply_init_unchained();
-    _tokenIdCounter = 0;
+    tokenIdCounter = 0;
     name = "Removal";
   }
 
+  function initializeRestrictedNORI(address restrictedNORIAddress)
+    external
+    onlyRole(DEFAULT_ADMIN_ROLE)
+  {
+    _restrictedNori = RestrictedNORI(restrictedNORIAddress);
```

Comment:
really don't love that this has to happen in multiple places now since it makes it tricky in the future if we ever need to update this address in multiple places (e.g., would need to not only be updated here but also in the fifomarket)

Reply:
is this something that we'd consider solved by using ENS resolution on-chain so that we only ever have to update the ENS registry if the contract address changes? and the name (or node hash version of the name or whatever) lives statically in this contract?

Reply:
maybe we should make a ticket to look into using that pattern everywhere?  I feel like changing that should stay out of the scope of this PR?